src/niri.rs

```diff
@@ -1694,6 +1694,18 @@ impl Niri {
                 is_tty && !client.get_data::<ClientState>().unwrap().restricted
             });
         let activation_state = XdgActivationState::new::<State>(&display_handle);
+        event_loop
+            .insert_source(
+                Timer::from_duration(XDG_ACTIVATION_TOKEN_TIMEOUT),
+                |_, _, state| {
+                    state.niri.activation_state.retain_tokens(|_, token_data| {
+                        token_data.timestamp.elapsed() < XDG_ACTIVATION_TOKEN_TIMEOUT * 2
```

Comment:
Is this `* 2` necessary? As far as I understand, if the `< timeout` condition fails here, the token is already unusable, even later this event loop iteration.

Reply:
Something about giving it a grace period or something like this :shrug: 
Not actually needed atm, will remove it.

Reply:
If this gets more complex it might be a good idea to add a shared function to test token/data validity and use it here and in `request_activation`, but not sure this will really work out.